### DISCUSSION TITLE
Streamline the abbreviations by not using one json for each abbreviation

### DISCUSSION THREAD

**Discussion by sudhansubhushanmishra** [Njong392/Abbreve#331] [Ideas]
I'm wondering if we could use a single JSON or JSONC file and append to it in a format like
```jsonc
{
  // Sample structure
  abbreviations: [
     idk: {
        short: 'IDK', 
        long: 'I don't know', 
        context: '...' 
     } 
  ], 
}
```
This way it would be easier to do bidirectional search and also maybe give context to somebody completely unfamiliar. Maybe this would be easier with TypeScript or there are probably better ways, but please feel free to suggest and discuss.

Thanks and have a great day!

**Comment by Njong392** [comment]
There was recently an issue similar to this. You can check out #42 concerning the structure of the db. It is still open because I did take the thought into consideration. As for all the abbreviations in one file, that was the initial state of the app. As it got longer, it was more prone to conflicts etc. I do have plans of creating a database to store the stuff.

**Reply by sudhansubhushanmishra** [reply]
We could check if the key is there and only then add, but I find your point so valid! Thanks for a great site though

**Reply by Njong392** [reply]
And thank YOU for your contribution. Thanks for checking Abbreve out!
